### PR TITLE
40122

### PR DESCRIPTION
**Pull request by harshaellanki**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [40122](https://devops.dchbx.org/redmine/issues/40122) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @davidison |

### CODE REVIEW

File: components/notifier/app/models/notifier/builders/employee_profile.rb [hunk 1]
```diff
@@ -219,15 +219,17 @@ def special_enrollment_period_start_on
...
+      end
     end
 
     def special_enrollment_period_submitted_at
       merge_model.special_enrollment_period.submitted_at = format_date(special_enrollment_period.submitted_at) if special_enrollment_period.submitted_at.present?
     end
 
     def special_enrollment_period_reporting_deadline
-      deadline = payload['notice_params']['qle_reporting_deadline']
+      deadline = payload['notice_params'] && payload['notice_params']['qle_reporting_deadline']
```
Comment (jayreddy519) [inline components/notifier/app/models/notifier/builders/employee_profile.rb]: Added missing params.

File: components/notifier/app/models/notifier/builders/employee_profile.rb [hunk 2]
```diff
@@ -200,16 +200,16 @@ def special_enrollment_period
...
     end
 
     def special_enrollment_period_title
-      merge_model.special_enrollment_period.title = special_enrollment_period.nil? ? payload['notice_params']['qle_title'] : special_enrollment_period.title
+      merge_model.special_enrollment_period.title = (payload['notice_params'] && payload['notice_params']['qle_title']) ? payload['notice_params']['qle_title'] : special_enrollment_period.title
     end
 
     def special_enrollment_period_qle_reported_on
-      merge_model.special_enrollment_period.qle_reported_on = (special_enrollment_period.present? && special_enrollment_period.qle_on.present?) ? format_date(special_enrollment_period.qle_on) : format_date(TimeKeeper.date_of_record)
+      merge_model.special_enrollment_period.qle_reported_on = format_date(TimeKeeper.date_of_record)
```
Comment (jayreddy519) [inline components/notifier/app/models/notifier/builders/employee_profile.rb]: Refactored special_enrollment_period_qle_reported_on

File: app/models/observers/notice_observer.rb [hunk 3]
```diff
@@ -345,8 +343,7 @@ def trigger_low_enrollment_notice(current_date)
       organizations_for_low_enrollment_notice(current_date).each do |organization|
         begin
           plan_year = organization.employer_profile.plan_years.where(:aasm_state.in => ["enrolling", "renewing_enrolling"]).first
-          #exclude congressional employees
-          next if ((plan_year.benefit_groups.any?{|bg| bg.is_congress?}) || (plan_year.effective_date.yday == 1))
+          next if plan_year.effective_date.yday == 1
```
Comment (jayreddy519) [inline app/models/observers/notice_observer.rb]: Allow Congressional groups to receive notices.

File: components/notifier/app/models/notifier/builders/employee_profile.rb [hunk 4]
```diff
@@ -204,6 +204,10 @@ def special_enrollment_period_event_on
       merge_model.special_enrollment_period.event_on = event_on
     end
 
+    def future_sep?
+      Date.strptime(special_enrollment_period_event_on, '%m/%d/%Y').future?
+    end
+
```
Comment (jayreddy519) [inline components/notifier/app/models/notifier/builders/employee_profile.rb]: New method added to check future sep date.

**Comment by Battula** [comment]
Branch is messed up. Closing the PR, will be addressed via #2680

**Pull request closed without merging**